backend/infrahub/core/validators/relationship/peer.py

```diff
@@ -125,3 +127,149 @@ async def check(self, request: SchemaConstraintValidatorRequest) -> list[Grouped
             await query.execute(db=self.db)
             grouped_data_paths_list.append(await query.get_paths())
         return grouped_data_paths_list
+
+
+class RelationshipPeerParentValidatorQuery(RelationshipSchemaValidatorQuery):
+    name = "relationship_constraints_peer_parent_validator"
+
+    def __init__(
+        self,
+        relationship: RelationshipSchema,
+        parent_relationship: RelationshipSchema,
+        peer_parent_relationship: RelationshipSchema,
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+
+        self.params["parent_relationship_id"] = parent_relationship.identifier
+        self.params["peer_parent_relationship_id"] = peer_parent_relationship.identifier
+        self.params["common_parent_relationship_id"] = relationship.identifier
+
+    async def query_init(self, db: InfrahubDatabase, **kwargs: dict[str, Any]) -> None:  # noqa: ARG002
+        branch_filter, branch_params = self.branch.get_query_filter_path(at=self.at.to_string(), is_isolated=False)
+        self.params.update(branch_params)
+
+        query = """
+        MATCH (n:%(node_kind)s)
+        CALL (n) {
+            MATCH path = (root:Root)<-[rroot:IS_PART_OF]-(n)
+            WHERE all(r in relationships(path) WHERE %(branch_filter)s)
+            RETURN path as full_path, n as active_node
+            ORDER BY rroot.branch_level DESC, rroot.from DESC
+            LIMIT 1
+        }
+        WITH full_path, active_node
+        WHERE all(r in relationships(full_path) WHERE r.status = "active")
+        MATCH path = (active_node)-[rrel1:IS_RELATED]-(rel:Relationship { name: $parent_relationship_id })-[rrel2:IS_RELATED]-(parent:Node)
+        WHERE all(r in relationships(path) WHERE %(branch_filter)s AND r.status = "active")
+        CALL (active_node) {
+            MATCH (active_node)-[:IS_RELATED]-(r:Relationship {name: $common_parent_relationship_id })-[:IS_RELATED]-(peer:Node)
+            WITH DISTINCT active_node, peer
+            MATCH (active_node)-[r1:IS_RELATED]-(r:Relationship {name: $common_parent_relationship_id })-[r2:IS_RELATED]-(peer:Node)
+            WHERE all(r in [r1, r2] WHERE %(branch_filter)s AND r.status = "active")
+            WITH peer, r1.status = "active" AND r2.status = "active" AS is_active
+            ORDER BY peer.uuid, r1.branch_level DESC, r2.branch_level DESC, r1.from DESC, r2.from DESC, is_active DESC
+            WITH peer, head(collect(is_active)) AS is_active
+            WHERE is_active = TRUE
+            RETURN peer
+        }
```

Comment:
you have to remove this `status = "active"` check so that you can get the latest deleted path from `active_node` to `peer` if it exists
this would be in the case of this peer being deleted on a branch

```suggestion
        CALL (active_node) {
            MATCH (active_node)-[:IS_RELATED]-(r:Relationship {name: $common_parent_relationship_id })-[:IS_RELATED]-(peer:Node)
            WITH DISTINCT active_node, peer
            MATCH (active_node)-[r1:IS_RELATED]-(r:Relationship {name: $common_parent_relationship_id })-[r2:IS_RELATED]-(peer:Node)
            WHERE all(r in [r1, r2] WHERE %(branch_filter)s)
            WITH peer, r1.status = "active" AND r2.status = "active" AS is_active
            ORDER BY peer.uuid, r1.branch_level DESC, r2.branch_level DESC, r1.from DESC, r2.from DESC, is_active DESC
            WITH peer, head(collect(is_active)) AS is_active
            WHERE is_active = TRUE
            RETURN peer
        }
```